src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -186,7 +186,7 @@ def load_arguments(self, _):
         c.argument('vm_name', options_list=['--vm-name'], id_part=None, completer=get_resource_name_completion_list('Microsoft.Compute/virtualMachines'))
         c.argument('new', action='store_true', help='create a new disk')
         c.argument('sku', arg_type=disk_sku, help='Underlying storage SKU')
-        c.argument('size_gb', options_list=['--size-gb', '-z'], help='size in GB. Max size: 4095 GB (certain preview disks can be larger).', type=int)
+        c.argument('size_gb', options_list=['--size-gb', '-z'], help='size in GB. Managed Disks Max size: 32767 GB, Unmanaged Disks Max size: 4095 GB', type=int)
```

Comment:
Please don't mention unmanaged disks here. `vm disk` is for managed disks.

Rather mention unmanaged disks [here](https://github.com/Azure/azure-cli/pull/8866/files#diff-046b3d363c785b7059452b6b72566229L256) under `vm unmanaged-disk attach`.

Also nit on my part:
```suggestion
        c.argument('size_gb', options_list=['--size-gb', '-z'], help='size in GB. Managed Disks Max size: 32767 GB. Unmanaged Disks Max size: 4095 GB', type=int)
```

Reply:
If you would like @yuemlu I can make these changes here and in #8858 in a single PR.